zcash.go

```diff
@@ -19,40 +24,10 @@ func init() {
 
 // ZCashDecodeToBytes converts the input string to a byte array
 func ZCashDecodeToBytes(input string) ([]byte, error) {
-	decoded, err := Base58ChecksumDecode(input, Base58DefaultAlphabet)
-	if err != nil {
-		return nil, err
-	}
-	err = checkIsValid(decoded)
-	if err != nil {
-		return nil, err
-	}
-	return decoded, nil
+	return Base58AddressDecodeToBytesPrefix(input, ZCashAddressLength, zcashPrefixes)
```

Comment:
we should decode to script like bitcoin: https://github.com/trustwallet/ens-coincodec/blob/master/bitcoin_test.go#L15

Reply:
Reworked as you suggested.  In fact ZCash is now a bitcoin fork, just like the others.